include/rmm/logger.hpp

```diff
@@ -58,13 +60,25 @@ struct logger_wrapper {
   {
     logger_.set_pattern("[%6t][%H:%M:%S:%f][%-6l] %v");
     logger_.flush_on(spdlog::level::warn);
+  }
 
+  void maybe_write_header()
+  {
+    if (logger_.should_log(spdlog::level::info) && not header_written_) {
 #ifdef CUDA_API_PER_THREAD_DEFAULT_STREAM
-    logger_.info("----- RMM LOG BEGIN [PTDS ENABLED] -----");
+      logger_.info("----- RMM LOG BEGIN [PTDS ENABLED] -----");
 #else
-    logger_.info("----- RMM LOG BEGIN [PTDS DISABLED] -----");
+      logger_.info("----- RMM LOG BEGIN [PTDS DISABLED] -----");
 #endif
-    logger_.flush();
+      logger_.flush();
+      header_written_ = true;
+    }
+  }
```

Comment:
There is a race condition here. Multiple threads could race to set `header_written_`. Instead of manually maintaining the initialization flag, can this logic just be put in the `logger_wrapper` ctor?